ui/__tests__/clientWorkout.test.js

```diff
@@ -0,0 +1,127 @@
+      ["date", false, "2023-09-23"],
+    ])("should sort by %s in %s order", (column, ascending, expected) => {
+      wrapper.setData({
+        clientWorkouts: clientWorkoutsData,
+        sortedColumn: column,
+        sortAscending: ascending,
+      });
+      expect(wrapper.vm.sortedWorkouts[0][column]).toBe(expected);
+    });
+  });
+
+  describe("Search functionality", () => {
+    test.each([
+      ["Chest Day", "workout_name", 1, "Chest Day"],
+      ["Calisthenics", "workout_name", 0],
+      ["cHeSt DaY", "workout_name", 1, "Chest Day"],
+      ["Intermediate", "difficulty", 1, "Intermediate"],
+      ["Intense", "notes", 1, "Intense"],
+    ])(
+      "should filter workouts by %s in %s column",
+      (query, column, expectedLength, expectedValue) => {
+        wrapper.setData({
+          clientWorkouts: clientWorkoutsData,
+          searchQuery: query,
+        });
+        expect(wrapper.vm.filteredWorkouts.length).toBe(expectedLength);
+        if (expectedLength > 0) {
+          expect(wrapper.vm.filteredWorkouts[0][column]).toBe(expectedValue);
+        }
+      }
+    );
+  });
+  describe("Pagination functionality", () => {
+    const clientWorkoutsData = [
+      {
+        id: 1,
+        workout_name: "Chest Day",
+        difficulty: "Intermediate",
+        date: "2023-09-23",
+        notes: "Intense",
+      },
+      {
+        id: 2,
+        workout_name: "Arm Day",
+        difficulty: "Beginner",
+        date: "2023-09-22",
+        notes: "Light",
+      },
+      {
+        id: 3,
+        workout_name: "Leg Day",
+        difficulty: "Advanced",
+        date: "2023-09-21",
+        notes: "Very Intense",
+      },
+    ];
+
+    test("should display correct number of workouts per page", () => {
+      wrapper.setData({
+        clientWorkouts: clientWorkoutsData,
+        currentPageAssignedWorkouts: 1,
+        itemsPerPageAssignedWorkouts: 2,
+      });
+      expect(wrapper.vm.paginatedAssignedWorkouts.length).toBe(2);
+      expect(wrapper.vm.paginatedAssignedWorkouts[0].workout_name).toBe(
+        "Chest Day"
+      );
+      expect(wrapper.vm.paginatedAssignedWorkouts[1].workout_name).toBe(
+        "Arm Day"
+      );
+    });
+
+    test("should calculate total pages correctly", () => {
+      wrapper.setData({
+        clientWorkouts: clientWorkoutsData,
+        itemsPerPageAssignedWorkouts: 2,
+      });
+      expect(wrapper.vm.totalPagesAssignedWorkouts).toBe(
+        Math.ceil(clientWorkoutsData.length / 2)
+      );
+    });
+
+    test("should navigate to next page correctly", () => {
+      wrapper.setData({
+        clientWorkouts: clientWorkoutsData,
+        currentPageAssignedWorkouts: 2,
+        itemsPerPageAssignedWorkouts: 2,
+      });
+      expect(wrapper.vm.paginatedAssignedWorkouts.length).toBe(1); // As we only have 3 workouts in the sample data
+      expect(wrapper.vm.paginatedAssignedWorkouts[0].workout_name).toBe(
+        "Leg Day"
+      );
+    });
```

Comment:
what happens if i have multiple pages and i sort by a column? Can you write a test for this?

Reply:
@zajonesck